### PR TITLE
Linting with ESLint

### PR DESCRIPTION
**Pull request by lelandjansen**
Adds the `ESLint` JavaScritpt linter to this project. The linting rules are based on ESLint's "recommended" settings and [Google's JavaScript style guide](https://google.github.io/styleguide/jsguide.html) with some custom rules specified in `package.json`. This PR also integrates `ESLint` with TravisCI meaning CI will fail if the code is not formatted according to these standards.

**Usage**
```
# Run linter
$ npm run lint
# Run linter with --fix option
$ npm run lint-fix
```

**Notes for reviewer**
* Please review `package.json` linting rules to make sure they align with your preferences.
* ~~Many JS style guides (including ESLint and Google) suggest camelCase instead of snake_case for variable and function names. It appears that this project uses snake_case so I've turned off the check. Is this what we want? Note that turning off the check does not enforce snake_case. [Docs](https://eslint.org/docs/rules/camelcase).~~
* ESLint also wants jsdocs by default but this project also doesn't seem to follow this convention (self-documenting?). I've also turned this off. Is this what we want? [Docs](https://eslint.org/docs/rules/require-jsdoc).
* Moved `browser.close();` to a `finally` block (previously unreachable, now is guaranteed to be run whether an exception is thrown or not).
* Will you please help me with CI? It looks like I don't have some of the environment variables required for the `openssl` decryption.

**TODO**
- [x] Play around with config settings
- [x] Fix existing failures (151 errors and counting...)
- [x] Integrate linting with TravisCI

### CODE REVIEW

**Comment by lelandjansen** [comment]
@nlieb Requesting review. Feel free to add me as an assignee to this PR.

**Comment by lelandjansen** [comment]
> Great work! I really like the consistency this will bring. Also thanks for catching that browser.close bug.

Thanks! np.

> I'd be more than happy to to switch to camelCase. You can do that in this PR if you are comfortable doing it or we can tackle it in a later PR.

Done. Tests pass locally, however, will you please do a thorough set of testing on your end? I want to make sure that the renaming didn't break anything.

Note: I refactored the JSON files to also be camelCase (following [Google's JSON style guide](https://google.github.io/styleguide/jsoncstyleguide.xml) and added a JSON plugin for ESLint. Note that the plugin doesn't enforce naming conventions so future reviewers will need to keep an eye out (unless you can find a better plugin).

> The only change I'd really like to see before merging: for the script section of the travis.yml file please note this: travis-ci/travis-ci#6188 We should be failing early on a lint failure.

Already done :) If you meant that we should fail all CI immediately if the linter fails here's my $0.02: I don't like to be forced to have perfect style while I'm trying to get code to work (i.e. get it working first, then clean up before submitting) and prefer to see all failures up-front. By joining all tests with an `&&` we will be forced to fix all style errors before discovering which tests fail. This can be especially frustrating if CI passes locally but is failing remotely. I'd recommend keeping the script as-is or having a separate Travis script for style-related checks. Your call.

Note: I did join the `openssl` and `npm test` commands with a `&&` since `npm test` is dependent on `openssl` successfully decrypting the file.

File: .travis.yml [hunk 1]
```diff
@@ -54,8 +54,15 @@ install:
   - yarn install
   # puppeteer's install script downloads Chrome
 script:
-  - openssl aes-256-cbc -K $encryption_key -iv $encryption_iv -in credentials_${TEST_UNIVERSITY_ID}.json.enc -out credentials.json -d
-  - npm test
+  - npm run lint
+  - |
```
Comment (nlieb) [inline .travis.yml]: Maybe it's my bash noobiness, but what does this pipe do? Google comes up with nothing

Comment (lelandjansen) [reply .travis.yml]: This is YAML syntax for a multiline literal string ([docs](http://yaml.org/spec/1.2/spec.html#id2795688)).

Comment (nlieb) [reply .travis.yml]: Ah cool, thanks!

**Comment by lelandjansen** [comment]
> Sadly I don't think there's a way to test this without adding the multi-term feature since I also have more than one term. Seeing as it currently doesn't work in light of this, I think we can merge it without guilt. Thoughts?

YOLO

**Pull request merged**